index.js

```diff
@@ -786,7 +786,10 @@
 
   //  Array$prototype$extend :: Array a ~> (Array a -> b) -> Array b
   function Array$prototype$extend(f) {
-    return [f(this)];
+    var xs = this;
+    return xs.map(function(_, i) {
+      return f(xs.slice(i));
+    });
```

Comment:
@safareli I'm adding an array one in the next day or so - spoken to @garyb about it. It's gonna be preeeetty ugly, but slicing seems to be the most performant way. :/